edge-util/src/Microsoft.Azure.Devices.Edge.Util/ConfigurationExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    public static class ConfigurationExtensions
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger(typeof(ConfigurationExtensions));
+
+        public static TimeSpan GetTimeSpan(this IConfiguration configuration, string key, TimeSpan defaultValue)
+        {
+            try
+            {
+                return configuration.GetValue(key, defaultValue);
+            }
+            catch (Exception ex)
+            when (IsParseException(ex))
+            {
+                Log.LogWarning(ex, $"Could not parse timespan for {key}");
+                return defaultValue;
+            }
+        }
+
+        static bool IsParseException(Exception ex)
+        {
+            return ex != null && (ex is OverflowException || ex is FormatException || IsParseException(ex.InnerException));
```

Comment:
Did you observe that the parse exception is sometimes (or always) wrapped in another exception, or is this theoretical?

Reply:
Observed

Reply:
Ah I see, Configuration wraps it in `InvalidOperationException`. We could simplify things by catching InvalidOperationException (no filtering), but this is fine too.